modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/AccountDetailsViewModel.kt

```diff
@@ -39,33 +35,15 @@ import timber.log.Timber
 @HiltViewModel
 class AccountDetailsViewModel @Inject constructor(
     userManager: UserManager,
-    private val subscriptionManager: SubscriptionManager,
+    private val paymentClient: PaymentClient,
     private val syncManager: SyncManager,
     private val settings: Settings,
     private val analyticsTracker: AnalyticsTracker,
     private val crashLogging: CrashLogging,
-    private val subscriptionMapper: SubscriptionMapper,
 ) : ViewModel() {
     internal val deleteAccountState = MutableStateFlow<DeleteAccountState>(DeleteAccountState.Empty)
     internal val signInState = userManager.getSignInState()
     internal val marketingOptIn = settings.marketingOptIn.flow
-    private val subscription = subscriptionManager.observeProductDetails().map { state ->
```

Comment:
This stream was pointless. We used it in the `showUpgradeBanner` stream but the logic was checking for `subsription != null` which is always `true` because the value was wrapped in `Optional`.